CRM/Civicase/Hook/BuildForm/DisplayAllCustomGroupsInCaseForm.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Display all custom groups in case form hook.
+ */
+class CRM_Civicase_Hook_BuildForm_DisplayAllCustomGroupsInCaseForm {
+
+  /**
+   * Adds all custom groups (inline/tabs) to the case form.
+   *
+   * @param CRM_Core_Form $from
+   *   CiviCRM Form
+   */
+  public function run ($form) {
+    if (!$this->shouldRun($form)) {
+      return;
+    }
+
+    $caseCategory = $form->getVar('_type');
```

Comment:
The type here refers to the entity type the custom group extends and the `CRM_Custom_Form_CustomDataByType` is not peculiar to the case entity type alone, what if the form is ran for say other custom field entity types, then this hook will be ran for those entity types too.

I think we can safely do this in the shouldRun function
```php
get_class($form) === CRM_Custom_Form_CustomDataByType::class && $form->getVar('_type') == 'Case'
```
because all the case category extend the Case custom field entity.